[BUG] (bugfix-2.0.x) Compile fail with ESP3D_WIFISUPPORT enabled

### Bug Description

As of recently Marlin does not compile with ESP3D_WIFISUPPORT defined in Configuration_adv.h; upon failure it prints the message `fatal error: ESPAsyncTCP.h: No such file or directory`.

### My Configurations

Configuration is completely default except for the aforementioned single change, fresh from bugfix-2.0.x branch. If needed a fork is [here](https://github.com/notabucketofspam/Marlin-ASM1/tree/potential-wifi-bug).

### Steps to Reproduce

Besides compiling with the single change, nothing else has to be done to reproduce the issue.

**Actual behavior:**
<details>
<summary>PIO task build terminal output</summary>
<p>

```
Verbose mode can be enabled via `-v, --verbose` option
Getting g++ path from cache
avr-g++.exe -D__MARLIN_FIRMWARE__ -w -dM -E -x c++ buildroot/share/PlatformIO/scripts/common-dependencies.h
Adding lib_deps for (ESP3D_)?WIFISUPPORT... 
Ignoring libs for (ESP3D_)?WIFISUPPORT... 
Ignoring libs: ['ESPAsyncTCP', 'U8glib-HAL', 'SlowSoftI2CMaster', 'TMC26XStepper', 'LiquidCrystal', 'SailfishLCD', 'lvgl', 'Adafruit MAX31865 library', 'Adafruit NeoPixel', 'Arduino-L6470', 'LiquidTWI2', 'TMCStepper']
CONFIGURATION: https://docs.platformio.org/page/boards/atmelavr/megaatmega2560.html
PLATFORM: Atmel AVR 2.2.0 > Arduino Mega or Mega 2560 ATmega2560 (Mega 2560)
HARDWARE: ATMEGA2560 16MHz, 8KB RAM, 248KB Flash
DEBUG: Current (simavr) On-board (simavr)
PACKAGES:
 - framework-arduino-avr 5.0.0
 - toolchain-atmelavr 1.50400.190710 (5.4.0)
Converting Marlin.ino
LDF: Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 10 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <ESP Async WebServer> 1.2.3
|-- <ESP3DLib> 1.0.0 #fd0f7b6
|   |-- <ESP32SSPD> 1.1.0 #36420ca
|   |-- <WebSockets> 2.2.1 #b3c5348
|   |   |-- <SPI> 1.0
|-- <WebSockets> 2.2.1 #b3c5348
|   |-- <SPI> 1.0
|-- <ESP32SSPD> 1.1.0 #36420ca
|-- <Wire> 1.0
|-- <SPI> 1.0
|-- <EEPROM> 2.0
|-- <SoftwareSerial> 1.0
Building in release mode
Compiling .pio\build\mega2560\src\src\HAL\AVR\HAL.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\AVR\HAL_SPI.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\AVR\MarlinSerial.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\AVR\Servo.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\AVR\eeprom.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\AVR\fast_pwm.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\AVR\fastio.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\AVR\u8g_com_HAL_AVR_sw_spi.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\AVR\watchdog.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\HAL_spi_L6470.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\backtrace\backtrace.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\backtrace\unwarm.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\backtrace\unwarm_arm.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\backtrace\unwarm_thumb.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\backtrace\unwarmbytab.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\backtrace\unwarmmem.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\backtrace\unwinder.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\backtrace\unwmemaccess.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\eeprom_api.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\eeprom_if_i2c.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\eeprom_if_spi.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\esp_wifi.cpp.o
Compiling .pio\build\mega2560\src\src\HAL\shared\servo.cpp.o
Compiling .pio\build\mega2560\src\src\MarlinCore.cpp.o
Compiling .pio\build\mega2560\src\src\core\multi_language.cpp.o
Compiling .pio\build\mega2560\src\src\core\serial.cpp.o
Compiling .pio\build\mega2560\src\src\core\utility.cpp.o
Compiling .pio\build\mega2560\src\src\feature\babystep.cpp.o
Compiling .pio\build\mega2560\src\src\feature\backlash.cpp.o
Compiling .pio\build\mega2560\src\src\feature\baricuda.cpp.o
Compiling .pio\build\mega2560\src\src\feature\bedlevel\bedlevel.cpp.o
Compiling .pio\build\mega2560\src\src\feature\binary_protocol.cpp.o
Compiling .pio\build\mega2560\src\src\feature\bltouch.cpp.o
Compiling .pio\build\mega2560\src\src\feature\cancel_object.cpp.o
Compiling .pio\build\mega2560\src\src\feature\caselight.cpp.o
Compiling .pio\build\mega2560\src\src\feature\closedloop.cpp.o
Compiling .pio\build\mega2560\src\src\feature\controllerfan.cpp.o
Compiling .pio\build\mega2560\src\src\feature\direct_stepping.cpp.o
Compiling .pio\build\mega2560\src\src\feature\e_parser.cpp.o
Compiling .pio\build\mega2560\src\src\feature\encoder_i2c.cpp.o
Compiling .pio\build\mega2560\src\src\feature\fanmux.cpp.o
Compiling .pio\build\mega2560\src\src\feature\filwidth.cpp.o
Compiling .pio\build\mega2560\src\src\feature\fwretract.cpp.o
Compiling .pio\build\mega2560\src\src\feature\host_actions.cpp.o
Compiling .pio\build\mega2560\src\src\feature\hotend_idle.cpp.o
Compiling .pio\build\mega2560\src\src\feature\joystick.cpp.o
Compiling .pio\build\mega2560\src\src\feature\max7219.cpp.o
Compiling .pio\build\mega2560\src\src\feature\mixing.cpp.o
Compiling .pio\build\mega2560\src\src\feature\mmu2\mmu2.cpp.o
Compiling .pio\build\mega2560\src\src\feature\pause.cpp.o
Compiling .pio\build\mega2560\src\src\feature\power.cpp.o
Compiling .pio\build\mega2560\src\src\feature\power_monitor.cpp.o
Compiling .pio\build\mega2560\src\src\feature\powerloss.cpp.o
Compiling .pio\build\mega2560\src\src\feature\probe_temp_comp.cpp.o
Compiling .pio\build\mega2560\src\src\feature\runout.cpp.o
Compiling .pio\build\mega2560\src\src\feature\snmm.cpp.o
Compiling .pio\build\mega2560\src\src\feature\solenoid.cpp.o
Compiling .pio\build\mega2560\src\src\feature\spindle_laser.cpp.o
Compiling .pio\build\mega2560\src\src\feature\tmc_util.cpp.o
Compiling .pio\build\mega2560\src\src\feature\touch\xpt2046.cpp.o
Compiling .pio\build\mega2560\src\src\feature\twibus.cpp.o
Compiling .pio\build\mega2560\src\src\feature\z_stepper_align.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\bedlevel\G26.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\bedlevel\G35.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\bedlevel\G42.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\bedlevel\M420.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\calibrate\G28.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\calibrate\G33.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\calibrate\G34_M422.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\calibrate\G425.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\calibrate\G76_M871.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\calibrate\M100.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\calibrate\M12.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\calibrate\M425.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\calibrate\M48.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\calibrate\M665.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\calibrate\M666.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\calibrate\M852.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M200-M205.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M217.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M218.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M220.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M221.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M281.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M301.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M302.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M304.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M305.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M43.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M540.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M575.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M672.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\config\M92.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M108_M112_M410.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M111.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M120_M121.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M17_M18_M84.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M211.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M226.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M280.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M3-M5.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M350_M351.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M380_M381.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M400.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M42.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M605.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M7-M9.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M80_M81.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M85.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M997.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\M999.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\control\T.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\eeprom\M500-M504.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\L6470\M122.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\L6470\M906.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\L6470\M916-918.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\advance\M900.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\baricuda\M126-M129.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\camera\M240.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\cancel\M486.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\caselight\M355.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\clean\G12.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\controllerfan\M710.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\digipot\M907-M910.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\filwidth\M404-M407.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\fwretract\G10_G11.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\fwretract\M207-M209.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\i2c\M260_M261.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\leds\M150.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\leds\M7219.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\macro\M810-M819.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\mixing\M163-M165.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\mixing\M166.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\pause\G27.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\pause\G60.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\pause\G61.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\pause\M125.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\pause\M600.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\pause\M603.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\pause\M701_M702.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\power_monitor\M430.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\powerloss\M1000.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\powerloss\M413.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\prusa_MMU2\M403.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\feature\runout\M412.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\gcode.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\geometry\G17-G19.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\geometry\G53-G59.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\geometry\G92.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\geometry\M206_M428.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\host\M110.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\host\M113.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\host\M114.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\host\M115.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\host\M118.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\host\M119.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\host\M16.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\host\M360.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\host\M876.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\lcd\M0_M1.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\lcd\M117.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\lcd\M145.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\lcd\M250.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\lcd\M300.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\lcd\M73.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\motion\G0_G1.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\motion\G2_G3.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\motion\G4.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\motion\G5.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\motion\G6.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\motion\G80.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\motion\M290.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\parser.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\probe\G30.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\probe\G31_G32.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\probe\G38.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\probe\M401_M402.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\probe\M851.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\probe\M951.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\queue.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\scara\M360-M364.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M1001.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M20.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M21_M22.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M23.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M24_M25.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M26.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M27.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M28_M29.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M30.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M32.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M33.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M34.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M524.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\sd\M928.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\stats\M31.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\stats\M75-M78.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\temp\M104_M109.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\temp\M105.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\temp\M106_M107.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\temp\M140_M190.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\temp\M141_M191.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\temp\M155.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\temp\M303.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\units\G20_G21.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\units\M149.cpp.o
Compiling .pio\build\mega2560\src\src\gcode\units\M82_M83.cpp.o
Compiling .pio\build\mega2560\src\src\lcd\extui\ui_api.cpp.o
Compiling .pio\build\mega2560\src\src\lcd\extui_anycubic_tft.cpp.o
Compiling .pio\build\mega2560\src\src\lcd\extui_dgus_lcd.cpp.o
Compiling .pio\build\mega2560\src\src\lcd\extui_example.cpp.o
Compiling .pio\build\mega2560\src\src\lcd\extui_malyan_lcd.cpp.o
Compiling .pio\build\mega2560\src\src\lcd\fontutils.cpp.o
Compiling .pio\build\mega2560\src\src\lcd\lcdprint.cpp.o
Compiling .pio\build\mega2560\src\src\lcd\ultralcd.cpp.o
Compiling .pio\build\mega2560\src\src\libs\BL24CXX.cpp.o
Compiling .pio\build\mega2560\src\src\libs\L64XX\L64XX_Marlin.cpp.o
Compiling .pio\build\mega2560\src\src\libs\buzzer.cpp.o
Compiling .pio\build\mega2560\src\src\libs\crc16.cpp.o
Compiling .pio\build\mega2560\src\src\libs\heatshrink\heatshrink_decoder.cpp.o
Compiling .pio\build\mega2560\src\src\libs\hex_print_routines.cpp.o
Compiling .pio\build\mega2560\src\src\libs\least_squares_fit.cpp.o
Compiling .pio\build\mega2560\src\src\libs\nozzle.cpp.o
Compiling .pio\build\mega2560\src\src\libs\numtostr.cpp.o
Compiling .pio\build\mega2560\src\src\libs\stopwatch.cpp.o
Compiling .pio\build\mega2560\src\src\libs\vector_3.cpp.o
Compiling .pio\build\mega2560\src\src\module\configuration_store.cpp.o
Compiling .pio\build\mega2560\src\src\module\delta.cpp.o
Compiling .pio\build\mega2560\src\src\module\endstops.cpp.o
Compiling .pio\build\mega2560\src\src\module\motion.cpp.o
Compiling .pio\build\mega2560\src\src\module\planner.cpp.o
Compiling .pio\build\mega2560\src\src\module\planner_bezier.cpp.o
Compiling .pio\build\mega2560\src\src\module\printcounter.cpp.o
Compiling .pio\build\mega2560\src\src\module\probe.cpp.o
Compiling .pio\build\mega2560\src\src\module\scara.cpp.o
Compiling .pio\build\mega2560\src\src\module\servo.cpp.o
Compiling .pio\build\mega2560\src\src\module\stepper.cpp.o
Compiling .pio\build\mega2560\src\src\module\stepper\L64xx.cpp.o
Compiling .pio\build\mega2560\src\src\module\stepper\TMC26X.cpp.o
Compiling .pio\build\mega2560\src\src\module\stepper\indirection.cpp.o
Compiling .pio\build\mega2560\src\src\module\stepper\trinamic.cpp.o
Compiling .pio\build\mega2560\src\src\module\temperature.cpp.o
Compiling .pio\build\mega2560\src\src\module\tool_change.cpp.o
Compiling .pio\build\mega2560\src\src\sd\Sd2Card.cpp.o
Compiling .pio\build\mega2560\src\src\sd\SdBaseFile.cpp.o
Compiling .pio\build\mega2560\src\src\sd\SdFatUtil.cpp.o
Compiling .pio\build\mega2560\src\src\sd\SdFile.cpp.o
Compiling .pio\build\mega2560\src\src\sd\SdVolume.cpp.o
Compiling .pio\build\mega2560\src\src\sd\cardreader.cpp.o
Compiling .pio\build\mega2560\lib414\ESP Async WebServer\AsyncEventSource.cpp.o
Compiling .pio\build\mega2560\lib414\ESP Async WebServer\AsyncWebSocket.cpp.o
In file included from C:\Users\Onion\.platformio\lib\ESP Async WebServer\src\AsyncEventSource.cpp:21:0:
C:\Users\Onion\.platformio\lib\ESP Async WebServer\src\AsyncEventSource.h:27:25: fatal error: ESPAsyncTCP.h: No such file or directory

*********************************************************************
* Looking for ESPAsyncTCP.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:ESPAsyncTCP.h"
* Web  > https://platformio.org/lib/search?query=header:ESPAsyncTCP.h
*
*********************************************************************

compilation terminated.
Compiling .pio\build\mega2560\lib414\ESP Async WebServer\SPIFFSEditor.cpp.o
In file included from C:\Users\Onion\.platformio\lib\ESP Async WebServer\src\AsyncWebSocket.cpp:22:0:
C:\Users\Onion\.platformio\lib\ESP Async WebServer\src\AsyncWebSocket.h:29:25: fatal error: ESPAsyncTCP.h: No such file or directory

*********************************************************************
* Looking for ESPAsyncTCP.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:ESPAsyncTCP.h"
* Web  > https://platformio.org/lib/search?query=header:ESPAsyncTCP.h
*
*********************************************************************

compilation terminated.
*** [.pio\build\mega2560\lib414\ESP Async WebServer\AsyncWebSocket.cpp.o] Error 1
*** [.pio\build\mega2560\lib414\ESP Async WebServer\AsyncEventSource.cpp.o] Error 1
In file included from C:\Users\Onion\.platformio\lib\ESP Async WebServer\src\SPIFFSEditor.h:3:0,
                 from C:\Users\Onion\.platformio\lib\ESP Async WebServer\src\SPIFFSEditor.cpp:1:
C:\Users\Onion\.platformio\lib\ESP Async WebServer\src/ESPAsyncWebServer.h:26:22: fatal error: functional: No such file or directory
compilation terminated.
*** [.pio\build\mega2560\lib414\ESP Async WebServer\SPIFFSEditor.cpp.o] Error 1
```

</p>
</details>

#### Additional Information

Last working compile was before [this commit](https://github.com/MarlinFirmware/Marlin/commit/6027055695f6fa49b857bec0552bf362e0b2bb56). I'm not sure how relevant this is but another user had similar include errors recently (#18739). Neither installing the ESPAsyncTCP library from [here](https://platformio.org/lib/show/305/ESPAsyncTCP) nor commenting `lib_ignore=ESPAsyncTCP` in platformio.ini will fix the issue.

--- a/Marlin/src/inc/SanityCheck.h
+++ b/Marlin/src/inc/SanityCheck.h
@@ -3042,5 +3042,5 @@ static_assert(   _ARR_TEST(3,0) && _ARR_TEST(3,1) && _ARR_TEST(3,2)
  * Sanity check for WIFI
  */
 #if ENABLED(ESP3D_WIFISUPPORT) && DISABLED(ARDUINO_ARCH_ESP32) 
-    #error "ESP3D_WIFISUPPORT library is only for ESP32 based controllers. Is not needed for standalone ESP3D modules."
+  #error "ESP3D_WIFISUPPORT requires an ESP32 controller. Use WIFISUPPORT for standalone ESP3D modules."
 #endif

--- a/Marlin/src/inc/SanityCheck.h
+++ b/Marlin/src/inc/SanityCheck.h
@@ -3037,3 +3037,10 @@ static_assert(   _ARR_TEST(3,0) && _ARR_TEST(3,1) && _ARR_TEST(3,2)
     #error "DIRECT_STEPPING is incompatible with LIN_ADVANCE. Enable in external planner if possible."
   #endif
 #endif
+
+/**
+ * Sanity check for WIFI
+ */
+#if ENABLED(ESP3D_WIFISUPPORT) && DISABLED(ARDUINO_ARCH_ESP32) 
+    #error "ESP3D_WIFISUPPORT library is only for ESP32 based controllers. Is not needed for standalone ESP3D modules."
+#endif

